src/libraries/System.Text.Json/System.Text.Json.SourceGeneration/ReflectionUtils/MetadataLoadContext.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis;
+
+namespace System.Reflection
+{
+    public class MetadataLoadContext
+    {
+        private readonly Dictionary<string, IAssemblySymbol> _assemblies = new Dictionary<string, IAssemblySymbol>(StringComparer.OrdinalIgnoreCase);
+        private readonly Compilation _compilation;
+
+        public MetadataLoadContext(Compilation compilation)
+        {
+            _compilation = compilation;
+            var assemblies = compilation.References
+                                        .OfType<PortableExecutableReference>()
+                                        .ToDictionary(r => AssemblyName.GetAssemblyName(r.FilePath),
+                                                      r => (IAssemblySymbol)compilation.GetAssemblyOrModuleSymbol(r)!);
+
+            foreach (var item in assemblies)
+            {
+                // REVIEW: We need to figure out full framework
+                // _assemblies[item.Key.FullName] = item.Value;
```

Comment:
Can you file an issue for this?

cc @davidfowl

Reply:
Will leave this unresolved until we create the project board and create an issue hopefully tomorrow.

Reply:
issue created: https://github.com/dotnet/runtimelab/issues/28